src/test/ui/dst/issue-90528-unsizing-suggestion-1.rs

```diff
@@ -0,0 +1,28 @@
+// Issue #90528: provide helpful suggestions when a trait bound is unsatisfied
```

Comment:
nit: you could probably collapse these tests down into fewer files - perhaps cases where a suggestion is made and where a suggestion isn't made - and then you can add `// run-rustfix` to the test to confirm the suggestion works.

Reply:
Sure. What should I do about the scenarios where the suggestion is wrong (see anywhere I put `bad suggestion`)?

Reply:
We could split these out into another test, as long as the suggestion is `MaybeIncorrect` then it's fine that it's sometimes wrong, it'll just need to be in a different test if we want `// run-rustfix` to work on the cases that do have correct suggestions.

Reply:
I'm still new to rustfix. Will it apply MaybeIncorrect changes if the test is annotated with `// run-rustfix`? As written, my code does not know when its suggestion is MachineApplicable and when it is MaybeIncorrect, so I just always give MaybeIncorrect.

Reply:
I actually can't remember, if it isn't doing it when you're trying this locally then we can just approve this - apologies for the runaround.

Reply:
I think that *in tests* `MaybeIncorrect` ones do get applied. If they don't what we normally do is separate all the cases that can be fixed into its own file and another for the rest.

Reply:
@estebank looks like you're right: https://rustc-dev-guide.rust-lang.org/tests/adding.html#other-header-commands

> `rustfix-only-machine-applicable` is equivalent to `run-rustfix` except it will only apply `MachineApplicable` suggestions. `run-rustfix` will apply all suggestions.